### ISSUE TITLE
Annotation name not found

### ISSUE THREAD

**Issue by rkyger-git** [mortazavilab/TALON#112]
Hello fairliereese, 

I ran into the same issue in this closed thread (https://github.com/mortazavilab/TALON/issues/91), when trying to run either **talon_filter_transcripts** or **talon_abundance**.
Neither my .db file, nor the output files from the previous step ("Running TALON"), are empty. 
Do you know what might be causing the problem?

_Originally posted by @rkyger-git in https://github.com/mortazavilab/TALON/issues/91#issuecomment-1306407923_

**Comment by fairliereese** [comment]
Could you please provide the calls you made to TALON and the filtering / abundance utilities that caused this error? Thanks!

**Comment by rkyger-git** [comment]
Yes, the code used was: 

**talon_initialize_database --f A103.gtf --g a103 --a a103_rna --o A103**

**talon --t 64 --f config.talon --db A103.db --build a103 --o A103_out**

**talon_filter_transcripts --db A103.db -a a103_rna --o A103_talon_filt_trans**

**Comment by fairliereese** [comment]
Sorry about the wait, could you please run this python code on your TALON database and tell me what the output is? I'd love to get to the bottom of this bug which seems to be common: 

```python
import sqlite3

database = <your talon database>

conn = sqlite3.connect(database)
cursor = conn.cursor()

cursor.execute("SELECT DISTINCT annot_name FROM gene_annotations")
annotations = [str(x[0]) for x in cursor.fetchall()]
conn.close()

print(annotations)
```

**Comment by rkyger-git** [comment]
Ok, I ran the code, the output I get is: `['TALON']`

**Comment by fairliereese** [comment]
Ok so I ran your `talon_initialize_database` command with a GTF I had lying around, and directly after running that the output, the annotations that are in the database are `['a100_rna']`. This leads me to believe something went wrong with the database initialization. Can you re-run the `talon_initialize_database` command and before running any reads through TALON, execute that python snippet again?

**Comment by rkyger-git** [comment]
I re-ran the `talon_initialize_database` command, and then ran the python snippet, the output I get this time is: `[]`.

**Comment by fairliereese** [comment]
Yeah so that's definitely unexpected and likely what's causing the issue. Would you be able to try re-installing the latest commits from GitHub and rerunning the `talon_initialize_database` command and seeing if that changes the annotations that are in your db? 

If that doesn't work you're welcome to send the GTF you're trying to initialize from to me (freese {at} uci.edu) and I will try to dig into what's going wrong in the source code.

**Comment by fairliereese** [comment]
Oh actually based on what I found out in [this issue](https://github.com/mortazavilab/TALON/issues/113), I think it's possible that your GTF might be formatted in a way that's incompatible with TALON. Check out [this wiki entry](https://github.com/mortazavilab/TALON/wiki/Formatting-a-GTF-annotation-to-work-with-TALON) for more details.

**Comment by rkyger-git** [comment]
Thanks, I reformatted the GTF, and I was able to get `talon_abundance` to work. However, `talon_filter_transcripts`, produces an empty output file, and no error messages.

**Comment by fairliereese** [comment]
Did you rerun `talon_initialize_database` and `talon` itself after formatting your GTF? Just checking.

**Comment by rkyger-git** [comment]
Yes
